addons/web/static/src/views/fields/signature/name_and_signature.js

```diff
@@ -0,0 +1,358 @@
+/** @odoo-module **/
+
+import { isMobileOS } from "@web/core/browser/feature_detection";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { useService } from "@web/core/utils/hooks";
+import { pick } from "@web/core/utils/objects";
+import { renderToString } from "@web/core/utils/render";
+import { getDataURLFromFile } from "@web/views/fields/file_handler";
+
+const { Component, useState, onWillStart, useRef, useEffect } = owl;
+
+export class NameAndSignature extends Component {
+    setup() {
+        this.rpc = useService("rpc");
+
+        this.htmlId = _.uniqueId();
+        this.defaultName = this.props.signature.name || "";
+        this.defaultFont = this.props.defaultFont || "";
+        this.fontColor = this.props.fontColor || "DarkBlue";
+        this.displaySignatureRatio = this.props.displaySignatureRatio || 3.0;
+        this.signatureType = this.props.signatureType || "signature";
+        this.noInputName = this.props.noInputName || false;
+        this.currentFont = 0;
+        this.drawTimeout = null;
+
+        this.state = useState({
+            signMode:
+                this.props.mode || (this.props.noInputName && !this.defaultName ? "draw" : "auto"),
+            showSignatureArea: this.noInputName || this.defaultName,
+            showFontList: false,
+            fontWidth: 0,
+        });
+
+        this.signNameInput = useRef("signNameInput");
+        const signInputLoad = useRef("signInputLoad");
+        useEffect(
+            (el) => {
+                if (el) {
+                    el.click();
+                }
+            },
+            () => [signInputLoad.el]
+        );
+
+        onWillStart(async () => {
+            this.fonts = await this.rpc(`/web/sign/get_fonts/${this.defaultFont}`);
+        });
+
+        const signature = useRef("signature");
+        useEffect(
+            (el) => {
+                if (el) {
+                    this.$signatureField = $(".o_web_sign_signature");
+                    this.$signatureField.on("change", () => {
+                        this.props.signature.isSignatureEmpty = this.isSignatureEmpty;
+                    });
+                    this.$signatureField.jSignature();
+                    this.resetSignature();
+                    this.props.signature.getSignatureImage = () =>
+                        this.$signatureField.jSignature("getData", "image");
+                    if (this.state.signMode === "auto") {
+                        this.drawCurrentName();
+                    }
+                }
+            },
+            () => [signature.el]
+        );
+    }
+
+    /**
+     * Draws the current name with the current font in the signature field.
+     */
+    drawCurrentName() {
+        const font = this.fonts[this.currentFont];
+        const text = this.getCleanedName();
+        const canvas = this.$signatureField.find("canvas")[0];
```

Comment:
would be nice not to have jquery usage all over the place (you probably only need it where you call jSignatue)